src/conn/conn_dhandle.c

```diff
@@ -554,8 +562,7 @@ __wt_conn_dhandle_discard_single(
 
 	dhandle = session->dhandle;
 
-	if (F_ISSET(dhandle, WT_DHANDLE_OPEN) ||
-	    (final && F_ISSET(dhandle, WT_DHANDLE_DEAD))) {
```

Comment:
Why was this changed?  It looked like it used to deal with discarding dead handles: how is that connected with a change to how statistics are allocated?